classes/Cookie.php

```diff
@@ -589,7 +589,7 @@ public function getSession($sessionId)
             $session = new CustomerSession($sessionId);
         }
 
-        if (!empty($session->getId())) {
+        if ($session && !empty($session->getId())) {
```

Comment:
In the error you got, it was like PHP put a default value on this variable. It seems that it could remain undefined in some cases, I wonder if 

```suggestion
        if (isset($session) && !empty($session->getId())) {
```

wouldn't be even better.